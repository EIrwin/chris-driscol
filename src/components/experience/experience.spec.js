// @flow
import React from 'react';
import renderer from 'react-test-renderer';
import Component from './experience.component';

let mockExperiences = 0;
const createMockExperience = () => {
  mockExperiences++;
  return {
    imageUrl: `https://image/url/${mockExperiences}`,
    duration: `duration ${mockExperiences}`,
    location: `location ${mockExperiences}`,
    title: `title ${mockExperiences}`,
    description: `description ${mockExperiences}`,
  };
};

const fixtures = {
  viewer: {},
  experiences: [
    createMockExperience(),
    createMockExperience(),
    createMockExperience(),
    createMockExperience(),
    createMockExperience(),
  ],
};

it('renders correctly', () => {
  const tree = renderer.create(<Component {...fixtures} />).toJSON();
  expect(tree).toMatchSnapshot();
});
