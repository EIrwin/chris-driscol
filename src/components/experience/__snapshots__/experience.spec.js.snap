// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<section
  className="home-section"
  id="experience"
>
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="text-center"
      >
        <h2
          className="section-heading"
        >
          My Experience
        </h2>
        <p
          className="section-subheading text-muted"
        >
          Here's what I've been up to..
        </p>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-lg-12"
      >
        <ul
          className="timeline"
        >
          <li
            className=""
          >
            <div
              className="timeline-image"
            >
              <img
                alt="location 1"
                className="img-circle img-responsive"
                src="https://image/url/1"
              />
            </div>
            <div
              className="timeline-panel"
            >
              <div
                className="timeline-heading"
              >
                <h4
                  className="timeline-year"
                >
                  duration 1
                </h4>
                <h5
                  className="subheading"
                >
                  location 1
                </h5>
              </div>
              <div
                className="timeline-body text-muted"
              >
                <p
                  className="text-muted"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "description 1",
                    }
                  }
                />
              </div>
            </div>
          </li>
          <li
            className="timeline-inverted"
          >
            <div
              className="timeline-image"
            >
              <img
                alt="location 2"
                className="img-circle img-responsive"
                src="https://image/url/2"
              />
            </div>
            <div
              className="timeline-panel"
            >
              <div
                className="timeline-heading"
              >
                <h4
                  className="timeline-year"
                >
                  duration 2
                </h4>
                <h5
                  className="subheading"
                >
                  location 2
                </h5>
              </div>
              <div
                className="timeline-body text-muted"
              >
                <p
                  className="text-muted"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "description 2",
                    }
                  }
                />
              </div>
            </div>
          </li>
          <li
            className=""
          >
            <div
              className="timeline-image"
            >
              <img
                alt="location 3"
                className="img-circle img-responsive"
                src="https://image/url/3"
              />
            </div>
            <div
              className="timeline-panel"
            >
              <div
                className="timeline-heading"
              >
                <h4
                  className="timeline-year"
                >
                  duration 3
                </h4>
                <h5
                  className="subheading"
                >
                  location 3
                </h5>
              </div>
              <div
                className="timeline-body text-muted"
              >
                <p
                  className="text-muted"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "description 3",
                    }
                  }
                />
              </div>
            </div>
          </li>
          <li
            className="timeline-inverted"
          >
            <div
              className="timeline-image"
            >
              <img
                alt="location 4"
                className="img-circle img-responsive"
                src="https://image/url/4"
              />
            </div>
            <div
              className="timeline-panel"
            >
              <div
                className="timeline-heading"
              >
                <h4
                  className="timeline-year"
                >
                  duration 4
                </h4>
                <h5
                  className="subheading"
                >
                  location 4
                </h5>
              </div>
              <div
                className="timeline-body text-muted"
              >
                <p
                  className="text-muted"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "description 4",
                    }
                  }
                />
              </div>
            </div>
          </li>
          <li
            className=""
          >
            <div
              className="timeline-image"
            >
              <img
                alt="location 5"
                className="img-circle img-responsive"
                src="https://image/url/5"
              />
            </div>
            <div
              className="timeline-panel"
            >
              <div
                className="timeline-heading"
              >
                <h4
                  className="timeline-year"
                >
                  duration 5
                </h4>
                <h5
                  className="subheading"
                >
                  location 5
                </h5>
              </div>
              <div
                className="timeline-body text-muted"
              >
                <p
                  className="text-muted"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "description 5",
                    }
                  }
                />
              </div>
            </div>
          </li>
          <li
            className="timeline-inverted timeline-final"
          >
            <div
              className="timeline-image"
            >
              <a
                className={undefined}
                onClick={[Function]}
                style={
                  Object {
                    "cursor": "pointer",
                  }
                }
              >
                <h4
                  className="hearmore"
                >
                  Want to
                  <br />
                  hear
                  <br />
                  more?
                </h4>
              </a>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>
</section>
`;
